lib/rbs/test/type_check.rb

```diff
@@ -340,7 +340,12 @@ def value(val, type)
         when Types::Variable
           true
         when Types::Literal
-          type.literal == val
+          begin
+            type.literal == val
+          rescue NoMethodError
+            raise if defined?(val.==)
+            false
+          end
```

Comment:
This is the exact same snippet as https://github.com/ruby/rbs/pull/2683/files#diff-b00b4cb204913354ffdc33b64788f911df698b99c4c65bf6cfa6bbea17e2870a... If either that or this PR is merged, i'll rebase the other